src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,182 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int elementsData;
+
+    public MyLinkedList() {
+        this.first = new Node<T>(null, null, null);;
+        this.last = new Node<T>(null, null, null);;
+        this.elementsData = 0;
+    }
+
+    public static void main(String[] args) {
+        MyLinkedList<String> a = new MyLinkedList<>();
+        a.add("1");
+        a.add("2");
+        a.add("3");
+        a.remove("3");
+        System.out.println(a);
+        System.out.println(a.size());
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        Node<T> node = new Node<T>(null, value, null);
+        if (elementsData == 0) {
+            this.first = node;
+            this.last = node;
+            this.elementsData += 1;
+            return true;
+        }
+        this.last.next = node;
+        node.prev = this.last;
+        node.next = null;
+        this.last = node;
+        this.elementsData += 1;
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > this.elementsData || index < 0) {
+            throw new IndexOutOfBoundsException("Index: " + index
+                    + " Size: " + this.elementsData);
+        }
+        Node<T> node = new Node<T>(null, value, null);
+        if (index == 0) {
+            this.first.prev = node;
+            node.next = this.first;
+            this.first = node;
+        }
+        if (index == this.elementsData) {
+            this.last.next = node;
+            node.prev = this.last;
+            this.last = node;
```

Comment:
add(value)

Reply:
did not deleted, because then i will should add if - else statesment in simple add